libraries/AP_Param/AP_Param.cpp

```diff
@@ -2197,6 +2197,22 @@ bool AP_Param::set_default_by_name(const char *name, float value)
     return false;
 }
 
+/*
+  set parameter defaults from a defaults_struct table
+  sends GCS message and panics (in SITL only) if parameter is not found
+ */
+void AP_Param::set_defaults_from_table(const struct defaults_table_struct *table, uint8_t count)
+{
+    for (uint8_t i=0; i<count; i++) {
+        if (!AP_Param::set_default_by_name(table[i].name, table[i].value)) {
+            gcs().send_text(MAV_SEVERITY_INFO, "set param default failure for %s", table[i].name);
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+            AP_HAL::panic("set param default failure for %s", table[i].name);
```

Comment:
I agree with @WickedShell that even for real vehicles this should stop loading. But I also agree with @rmackay9 that panic isn't great for users.

So I would recommend to replace GCS text and panic with calling `sensor_config_error` from AP_BoardConfig. The method name isn't great, but it's functionality is exactly what we want: stop booting while still allowing communication with the board and periodically giving an error message.